core/src/message/mod.rs

```diff
@@ -86,6 +87,62 @@ impl Payload {
     }
 }
 
+/// Buffer which size cannot be bigger then max allowed payload size.
+#[derive(Clone, Default, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct PanicBuffer(Payload);
+
+impl PanicBuffer {
+    /// Returns ref to the internal data.
+    pub fn inner(&self) -> &[u8] {
+        self.0.inner()
+    }
+
+    fn as_limited_str(&self) -> Option<LimitedStr> {
+        let s = core::str::from_utf8(self.0.inner()).ok()?;
+        LimitedStr::try_from(s).ok()
+    }
+}
+
+impl<T> From<T> for PanicBuffer
+where
+    T: AsRef<[u8]>,
+{
+    fn from(value: T) -> Self {
```

Comment:
shouldn't it be try_from?

Reply:
that we will handle on a backend? we can return nothing or unwrap in case of checked data_len ahead?